src/backend/distributed/sql/udfs/citus_job_status/latest.sql

```diff
@@ -0,0 +1,73 @@
+CREATE OR REPLACE FUNCTION pg_catalog.citus_job_status (
+    job_id bigint DEFAULT NULL
+)
+    RETURNS TABLE (
+            job_id bigint,
+            state pg_catalog.citus_job_status,
+            job_type name,
+            description text,
+            started_at timestamptz,
+            finished_at timestamptz,
+            details jsonb
+    )
+    LANGUAGE SQL
+    STRICT
+    AS $fn$
+    WITH rp AS (
```

Comment:
This can be pretty heavy to run and I doubt we have set the COST correctly, probably good to ensure it isn't run multiple times. Running it multiple times might also cause some weird inconsistencies
```suggestion
    WITH rp AS MATERIALIZED (
```

Reply:
This is a really nice piece of advice. I always assume that the table expressions are always materialized.